### PR TITLE
[GPU] Remove legacy fused ops

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Removed pass with legacy activation primitive fusion and added conversion during kernel selector params creation for single activation in fused ops
 -  Fix winograd kernel compilation

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp [hunk 1]
```diff
@@ -322,11 +320,11 @@ void prepare_buffer_fusing::run(program& p) {
         // The condition below check only output layout as cases like
         // (dyn_shape) -> reshape -> (static_shape) -> some_static_primitive
         // may have invalid set_arguments call as output memory of reshape won't be available until reshape primitive is executed
-        if (node->is_type<reshape>() && is_dynamic && is_planar && no_pad && !node->is_output() && node->get_fused_activations_funcs().empty()) {
+        if (node->is_type<reshape>() && is_dynamic && is_planar && no_pad && !node->is_output() && !node->has_fused_primitives()) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:323]: This assumes that rehsape can have "fused operation" but there is no "GetSupportedFusedOps" in reshape. Does it make sense?

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp [hunk 2]
```diff
@@ -322,11 +320,11 @@ void prepare_buffer_fusing::run(program& p) {
...
             return true;
         }
 
-        if (node->is_dynamic() || node->is_output() || (!node->get_fused_activations_funcs().empty())) {
+        if (node->is_dynamic() || node->is_output() || node->has_fused_primitives()) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:327]: Does this behave same? Previously it was excluding only "fused activation" but now it excludes all fused cases.

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:327]: I think missing fused_primitives check before is a bug. This pass handles concat,crop and reshape primitives and if those have any type of fused operation - they can't be optimized out.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp (hunk 1 above)
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:323]: For reshape we allow fusion for single activation via fused_ops mechanism and it's supposed to be converted later to legacy desc for jitter. So missing `GetSupportedFusedOps` impl is not a big issue. Also, we can try to prohibit all fusions into reshape later, as those seems to be inefficient (optimized out reshape + optimized activation can be faster than single reshape kernel)

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:323]: Is the idea to handle such cases later when detected by "GPU_DEBUG_LOG << activation_node.id() << " is fused by legacy conditions! Consider adding selected kernel with fused ops support" ?

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:323]: After I added the second comment I saw your response. Resolving.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**